src/stan_math_backend/Statement_gen.ml

```diff
@@ -108,34 +146,42 @@ let rec pp_statement (ppf : Format.formatter)
   | Block _ | SList _ | Decl _ | Skip | Break | Continue -> ()
   | _ -> Locations.pp_smeta ppf meta ) ;
   match pattern with
+  (* TUPLE MAYBE assigning to tuples
+     Right now, tuples are assigned with `=`.
+     There's a potential issue when the tuple has an element that would normally be assigned to with `assign()`. I'm not sure what sort of checks are done inside `assign`, but they currently aren't being called on e.g. matrices inside tuples.
+  *)
+  (* Use = for any tuple-valued assignment or any scalar-valued assignment without LHS indices *)
+  | Assignment
```

Comment:
Refactored assign vs = calls. I was trying to clarify/simplify the logic and found at least one bug, but this might be outdated now